src/Nethermind/Nethermind.Optimism/OPL1CostHelper.cs

```diff
@@ -145,109 +145,100 @@ public static UInt256 ComputeL1CostPreEcotone(UInt256 dataGasWithOverhead, UInt2
     // Based on:
     // https://github.com/ethereum-optimism/op-geth/blob/7c2819836018bfe0ca07c4e4955754834ffad4e0/core/types/rollup_cost.go
     // https://github.com/Vectorized/solady/blob/5315d937d79b335c668896d7533ac603adac5315/js/solady.js
-    [SkipLocalsInit]
     public static uint ComputeFlzCompressLen(Transaction tx)
     {
         byte[] encoded = Rlp.Encode(tx, RlpBehaviors.SkipTypedWrapping).Bytes;
 
-        [SkipLocalsInit]
         static uint FlzCompressLen(byte[] data)
         {
             uint n = 0;
-            uint[] ht = ArrayPool<uint>.Shared.Rent(8192);
-            try
+            Span<uint> ht = stackalloc uint[8192];
```

Comment:
Maybe add a comment about not using `SkipLocalsInit` as depending on values initalized to 0